### PR TITLE
12666 Fixed database list not being populated on connection dialog

### PR DESCRIPTION
**Pull request by JustinMDotNet**
Changed isPasswordrequried for Kusto. Passed azure account into onFetchDatabases and set on tempProfile.

This PR fixes # https://github.com/microsoft/azuredatastudio/issues/12666

### CODE REVIEW

File: src/sql/platform/connection/common/providerConnectionInfo.ts
Comment (Charles-Gagnon) [inline src/sql/platform/connection/common/providerConnectionInfo.ts:200]: Why is kusto special for this anyways? The kusto provider should be returning the correct value for isRequired I would think

File: src/sql/workbench/services/connection/browser/connectionManagementService.ts
Comment (JustinMDotNet) [inline src/sql/workbench/services/connection/browser/connectionManagementService.ts:316]: Changed this from OR to AND to only show the error dialog when there's no password and no token.

Comment (Charles-Gagnon) [reply src/sql/workbench/services/connection/browser/connectionManagementService.ts:316]: Why? If a password is required why would we have an auth token?

And how is this related to the linked issue?

Comment (JustinMDotNet) [reply src/sql/workbench/services/connection/browser/connectionManagementService.ts:316]: My understanding is that this function is attempting to connect to the server, This `if` check is seeing if it has a password and is required. In this case, the server doesn't have a password, but rather has an access token so we should connect with that.

File: src/sql/workbench/services/connection/browser/connectionManagementService.ts
Comment (Charles-Gagnon) [inline src/sql/workbench/services/connection/browser/connectionManagementService.ts:316]: Can you explain why this is necessary? When would we have a token credential AND require a password such that this is needed?

File: src/sql/workbench/services/connection/browser/connectionManagementService.ts
Comment (Charles-Gagnon) [reply src/sql/workbench/services/connection/browser/connectionManagementService.ts]: So I think the real issue here is that the isPasswordRequired doesn't take into account other auth types for non-MSSQL connections. Look at the check here for isPasswordRequired : https://github.com/microsoft/azuredatastudio/blob/main/src/sql/platform/connection/common/providerConnectionInfo.ts#L203

Note that we special case MSSQL to only set isPasswordRequired to true if the auth type is SqlAuth - since the other auth types don't require a password.

```
		let isPasswordRequired = optionMetadata.isRequired;
		if (this.providerName === Constants.mssqlProviderName) {
			isPasswordRequired = this.authenticationType === ProviderConnectionInfo.SqlAuthentication && optionMetadata.isRequired;
		}
```

So the actual fix here would be to allow the providers to specify whether a password is required for each auth type instead.

**Comment by JustinMDotNet** [comment]
This may also resolve: https://github.com/microsoft/azuredatastudio/issues/12843

**Review by Charles-Gagnon — APPROVED** [review]
It looks ok - but I still think this is not really the best way to handle this kind of logic because it basically means that core has to know about all the different authentication types and how they're stored. What happens when we have another extension added with a different auth type? Same kind of thing will happen. 

This also isn't really addressing the issue I brought up where isPasswordRequired isn't taking into account that a provider may have multiple different auth types with some requiring a password and some not.

Both of these are core issues and it would be nice to fix them instead of doing workaround solutions - but I'll leave that up to you and the core team to decide. 

I'd suggest chatting with @kburtram or someone else from there first before merging this in though.

**Review by alanrenmsft — CHANGES_REQUESTED** [review]
ooops, the commit history and changes should be cleaned up before we can review it, please fix it or create a new PR with just your changes.

**Comment by VasuBhog** [comment]
> This may also resolve: #12843

I tested the branch and changes to see if it resolves the issue, and it does not. There is still an issue with the default database for Kusto not properly being changed to another database. This is something to further investigate as on the notebook level we see that the database changes, but on the Kusto connection/querying there is an issue.

**Comment by JustinMDotNet** [comment]
Reopening PR https://github.com/microsoft/azuredatastudio/pull/13239

**Pull request closed without merging**
